### PR TITLE
feat(devices): notify other devices of a new device

### PR DESCRIPTION
**Pull request by eoger**
See #1229 for context and older discussion.

### CODE REVIEW

File: lib/push.js [hunk 1]
```diff
@@ -59,6 +71,24 @@ module.exports = function (log, db) {
...
 
     /**
+     *  Notifies all devices (except the one who joined) that a new device joined the account
+     *
+     * @param uid
+     * @param deviceName
+     * @param currentDeviceId
+     * @promise
+     */
+    notifyNewDevice: function notifyNewDevice(uid, deviceName, currentDeviceId) {
```
Comment (vladikoff) [inline lib/push.js]: @eoger if we are going to do https://github.com/mozilla/fxa-auth-server/issues/1139 in the future you might want to rethink the naming here. 

cc @rfk

Comment (rfk) [inline lib/push.js]: Sorry, I don't have a lot of ongoing context on this feature - naming of what exactly?

**Comment by eoger** [comment]
Added JSON schema validation.

**Comment by eoger** [comment]
I changed the wording from newDevice to deviceConnected (see https://github.com/mozilla/fxa-auth-server/pull/1250#discussion_r63439468)

**Comment by eoger** [comment]
@thomcc Mark told me you might be interested in the JSON validation

**Comment by vladikoff** [comment]
Is this missing an update shrinkwrap with `--dev`?

File: lib/push.js [hunk 2]
```diff
@@ -10,6 +10,11 @@ var ERR_DATA_BUT_NO_KEYS = 'Data payload present but missing key(s)'
 
 var LOG_OP_PUSH_TO_DEVICES = 'push.pushToDevices'
 
+var PUSH_PAYLOAD_SCHEMA_VERSION = 1
+var pushCommands = {
```
Comment (vladikoff) [inline lib/push.js]: nit: maybe make this `pushCommands` all uppercase to signal `const` without using it, to follow the content-server and other variables above.

File: lib/routes/account.js [hunk 3]
```diff
@@ -872,6 +875,7 @@ module.exports = function (
         db[operation](sessionToken.uid, sessionToken.tokenId, payload).then(
           function (device) {
             reply(butil.unbuffer(device))
+            push.notifyDeviceConnected(sessionToken.uid, device.name, device.id.toString('hex'))
```
Comment (vladikoff) [inline lib/routes/account.js]: 👍

**Comment by vladikoff** [comment]
2 nits, looks good otherwise!

**Comment by vladikoff** [comment]
Good Stuff!

**Pull request merged**